src/services/RewardService.ts

```diff
@@ -39,6 +39,12 @@ export default class RewardService {
             return { error: 'This reward already disabled by it owner' };
         }
 
+        // Can not claim if reward has an unlockDate and the Now is not greather than unlockDate
+        // (included pending withdrawars)
+        if (new Date().getTime()< reward.withdrawUnlockDate.getTime()) {
```

Comment:
@valeriagrazzini Also mind that you don't necessarily have to construct a Date to be able to call now():
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date/now